crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -680,12 +718,12 @@ impl RoomInfo {
 
     /// Mark this Room as having all the members synced
     pub fn mark_members_synced(&mut self) {
-        self.members_synced = true;
+        self.sync_info = SyncInfo::FullySynced;
```

Comment:
I think there's still a problem here. If we call any method to fetch members before we receive a sync response, we're going to transition from `NoState` into `FullySynced`. This isn't quite right, we're actually going to be in a middle state where we have all the members but no other state events.

I think we need another enum variant to represent this state and the `mark_members_synced` should transition either into the `FullySynced` or into the new state depending on the current state.

Furthermore, the logic that transitions the `SyncInfo` of the room, once you receive a sync response, seems to be completely missing. 

The PR is also not protecting against the sending of unencrypted events in encrypted rooms. That is, it doesn't check the `SyncInfo` when it is sending out an event, whether it needs to fetch the `m.room.encryption` state event from the server or if it can rely on the event in the store.

Reply:
Yeah, that as well as the upcoming sliding sync (which as _incomplete_ state additonally then) made me a headache. I went for the existing pattern when I took over but the three supposedly following states don't really make sense. I think we want to split this up into two separate states for members and states as they aren't necessarily linked directly.

I'll refactor that along with the fix for the breaking integration test.